web.browser/src/containers/PostList/PostListContainer.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { updateVote, sortPopular, sortNewest } from '../../redux/actions';
+import PostList from './PostList';
+
+import './styles.css';
+
+class PostListContainer extends Component {
+  // constructor() {
+  //   super();
+
+  //   this.updateVote = this.updateVote.bind(this);
+  //   this.sortPopular = this.sortPopular.bind(this);
+  //   this.sortNewest = this.sortNewest.bind(this);
+
+  //   this.state={
+  //     posts: data.posts,
+  //     orderby: 'newest'
+  //   }
+  // }
+
+  updateVote(post) {
+    const upVote = post.votes + 1;
+    this.props.dispatch(updateVote(upVote));
+    // let moreVotes = post.votes++;
+    // this.setState({
+    //   votes: moreVotes
+    // });
+  }
+
+  sortPopular() {
+    this.props.dispatch(sortPopular());
+    // let sortedPosts = posts.sort(function (a, b) {
+    //   return b.votes - a.votes;
+    // });
+    // this.setState({
+    //   orderby: 'popular',
+    //   posts: sortedPosts
+    // })
+  }
+
+  sortNewest() {
+    this.props.dispatch(sortNewest());
+    // let sortedPosts = posts.sort(function (a, b) {
+    //   return b.id - a.id;
+    // });
+    // this.setState({
+    //   orderby: 'newest',
+    //   posts: sortedPosts
+    // });
+  }
+
+
+  render() {
+    return (
+      <PostList
+        posts={this.props.posts}
+        updateVote={this.updateVote}
+        sortPopular={this.sortPopular}
+        sortNewest={this.sortNewest}
+      />
+    );
+  }
+}
+
+PostListContainer.propTypes = {
+  dispatch: PropTypes.func.isRequired
+};
+
+function mapStateToProps(state) {
+  return {
+    posts: state.posts,
+  };
+}
+
```

Comment:
Alright, you can actually delete 90% of the code in this contained by using [mapDispatchToProps](http://redux.js.org/docs/basics/UsageWithReact.html#implementing-container-components).

It's a function that returns an object, similar to `mapStateToProps`. But it gets the `dispatch` function as a parameter.

```js
const mapDispatchToProps = (dispatch) => {
  return {
    updateVote: () => {
      dispatch(updateVote());
    },
    sortPopular: () => {
      dispatch(sortPopular());
    },
    // ...etc
  }
}
```

Then, when you call connect, you also pass in `mapDispatchToProps`.

```js
export default connect(mapStateToProps, mapDispatchToProps)(PostListContainer);
```

Now you will have functions called `updateVote` and `sortPopular` in your props!